changelog.d/15068.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug introduced in Synapse 1.76.0 where partially-joined rooms could not be deleted using the [purge room API](https://matrix-org.github.io/synapse/latest/admin_api/rooms.html#delete-room-api).
```

Comment:
This was (presumably) present in earlier Synapse versions too, if you had the experiemental faster joins feature turned on. But I didn't want to mention that here.